engines/director/archive.cpp

```diff
@@ -1148,4 +1168,214 @@ Common::String RIFXArchive::formatArchiveInfo() {
+		if (it->tag != _metaTag &&
+			it->tag != SWAP_BYTES_32(_metaTag) &&
+			it->tag != MKTAG('i', 'm', 'a', 'p') &&
+			it->tag != MKTAG('m', 'm', 'a', 'p') &&
+			it->tag != MKTAG('K', 'E', 'Y', '*') 
+		) {
+			writeStream->seek(it->offset + 8);
+			writeStream->writeStream(getResource(it->tag, it->index));
+		}
+	}
+
+	Common::DumpFile out;
+	
+	// Write the movie out, stored in dumpData
+	if (out.open(path, true)) {
+		out.write(dumpData, _size);
+		out.flush();
+		out.close();
+		debugC(3, kDebugLoading, "RIFXArchive::writeStream:Saved the movie as file %s", path.toString().c_str());
+	} else {
+		warning("RIFXArchive::writeStream: Error saving the file %s", path.toString().c_str());
+	}
+
+	free(dumpData);
+	delete writeStream;
+	return true;
+}
+
+bool RIFXArchive::writeMemoryMap(Common::SeekableMemoryWriteStream *writeStream) {
+	writeStream->writeUint32LE(MKTAG('i', 'm', 'a', 'p')); // The "imap" resource
+	writeStream->writeUint32LE(_imapLength);		// length of "imap" resource
+	writeStream->writeUint32LE(_mapversion);		// "imap" version
+	writeStream->writeUint32LE(_mmapOffset);		// offset of the "mmap" resource
+	writeStream->writeUint32LE(_version);
+
+	writeStream->seek(_mmapOffset);
+	writeStream->writeUint32LE(MKTAG('m', 'm', 'a', 'p'));
+
+	uint32 newResCount = _resources.size();
+	
+	// Need to recalcualte the following things
+	// Similarl to the RIFX resource, we'll need to update the size of the mmap resource whenever there is some change
+	writeStream->writeUint32LE(getResourceSize(MKTAG('m', 'm', 'a', 'p'), 0));
+	writeStream->writeUint16LE(_mmapHeaderSize);
+	writeStream->writeUint16LE(_mmapEntrySize);
+	
+	writeStream->writeUint32LE(newResCount + _totalCount - _resCount); // _totalCount - _resCount is the number of empty entries
+	writeStream->writeUint32LE(newResCount);
+	writeStream->seek(8, SEEK_CUR);
+
+	// ID of the first 'free' resource
+	writeStream->writeUint32LE(0);
+
+	for (auto &it : _resources) {
+		debugC(3, kDebugLoading, "Writing RIFX Resource: tag: %s, size: %d, offset: %08x, flags: %x, unk1: %x, nextFreeResourceID: %d", 
+			tag2str(it->tag), it->size, it->offset, it->flags, it->unk1, it->nextFreeResourceID);
+		
+		// Write down the tag, the size and offset of the current resource
+		writeStream->writeUint32LE(it->tag);
+		writeStream->writeUint32LE(it->size);
+		writeStream->writeUint32LE(it->offset);
+		
+		// Currently ignoring flags, unk1 and nextFreeResourceID
+		writeStream->writeUint16LE(it->flags);
+		writeStream->writeUint16LE(it->unk1);
+		writeStream->writeUint32LE(it->nextFreeResourceID);
+	}
+
+	return true;
+}
+
+// This Function is incomplete, needs further changes before can be used
+bool RIFXArchive::writeAfterBurnerMap(Common::SeekableMemoryWriteStream *writeStream) {
+	warning("RIFXArchive::writeAfterBurnerMap: STUB: Incomplete function, needs further changes, AfterBurnerMap not written");
+	return false;
+
+	writeStream->writeUint32LE(MKTAG('F', 'v', 'e', 'r'));
```

Comment:
```suggestion
#if 0
	writeStream->writeUint32LE(MKTAG('F', 'v', 'e', 'r'));
```
Please wrap big chunks of code like this into `#if 0`, to avoid "unreachable code" compiler warning